File: LeetCode/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int n = nums.size();
        vector<int> res(n);
        int left = 0, right = n-1;
        int i = 0, j = n-1;

        while(i < n && j >= 0){
            if(nums[i] < pivot){
                res[left] = nums[i];
                left++;
            }
            if(nums[j] > pivot){
                res[right] = nums[j];
                right--;
            }
            i++, j--;
        }

        while(left <= right){
            res[left] = pivot;
            left++;
        }

        return res;
    }
};

File: LeetCode/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.go
func pivotArray(nums []int, pivot int) []int {
    n := len(nums)
    res := make([]int, n);
    left, right := 0, n-1;
    i, j := 0, n-1;
   
    for i < n && j >= 0{
        if nums[i] < pivot{
            res[left] = nums[i];
            left++;
        } 
        if nums[j] > pivot{
            res[right] = nums[j];
            right--;
        }
        i++;
        j--;
    }

    for left <= right{
        res[left] = pivot;
        left++;
    }

    return res;
}